File: src/VirtualNode.java
public class VirtualNode {
    Worker worker;
    String hash;
    int partition;

    //TODO: add previous virtual node field  (abort?)
    //TODO: add previous virtual node hash field (abort?)

    public VirtualNode(Worker worker, String hash, int partition){
        this.worker = worker;
        this.hash = hash;
        this.partition = partition;
    }
}

File: apps/lab3/lib/virtual_node.ex
defmodule Dynamo.VirtualNode do

  alias __MODULE__

  defstruct(
    worker: nil,
    hash: nil,
    partition: nil
  )

  #@spec new(atom(),
  #        atom(),
  #        non_neg_integer()) :: %VirtualNode{}
  def new(w, h, p) do
    IO.puts(5555555)
    %VirtualNode{
      worker: w,
      hash: h,
      partition: p
    }
  end
end